### PR TITLE
Add module-expression

### PR DESCRIPTION
**Pull request by sosukesuzuki**
# [View Rendered Text](https://github.com/sosukesuzuki/estree/blob/add-js-module-blocks/experimental/js-module-blocks.md)

Adds spec for [JS Module Blocks](https://github.com/tc39/proposal-js-module-blocks). (ref: https://github.com/babel/babel/pull/12469#discussion_r539682666)

### CODE REVIEW

**Review by nzakas — APPROVED** [review]
LGTM

File: experimental/js-module-blocks.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+## Expressions
+
+### ModuleExpression
+
+```js
+interface ModuleExpression <: Expression {
+    type: "ModuleExpression";
+    body: Program;
+}
+```
```
Comment (JLHwung) [inline experimental/js-module-blocks.md:12]: Consider add a note that the `sourceType` of `body` must be `"module"`.

**Review by JLHwung — APPROVED** [review]
LGTM with a nit.

**Review by nzakas — APPROVED** [review]

**Comment by sosukesuzuki** [comment]
https://github.com/babel/babel/pull/12469 has been merged and Babel 7.13 has been released! Can we merge this?

**Comment by JLHwung** [comment]
ping @RReverser

**Review by RReverser — APPROVED** [review]

File: experimental/js-module-blocks.md (hunk 1 above)
Comment (RReverser) [reply experimental/js-module-blocks.md:12]: We can probably subtype `Program` to a `Module` for that. But it's fine for experimental.

**Comment by fisker** [comment]
@nzakas Don't you think this will break losts of ESLint plugins? Many plugin listen on `Program` or `Program:exit` and assume it's the AST root.

**Comment by nzakas** [comment]
Yes, that’s a good point, which is why we probably want to go with @RReverser’s suggestion in the final version: https://github.com/estree/estree/pull/238#discussion_r582888224

**Pull request merged**